src/components/pie/index.js

```diff
@@ -90,14 +94,87 @@ export class PieChartComponent extends events.EventEmitter {
           }
         });
 
-        that.emit('loaded', that, data);
         that.emit('ready', that, data, null);
       })
       .catch((error) => {
         that.emit('error', that, error);
         that.emit('ready', that, null, error);
       });
   }
+
+  _getData(endpoint, cube, params, maxSlices) {
+    var that = this;
+    that.emit('loading', that);
+
+    var api = this.getApiInstance();
+    return api.aggregate(endpoint, cube, params)
+      .then((data) => that._groupSlicesIfMoreThan(data, maxSlices))
+      .then((data) => {
+        that.emit('loaded', that, data);
+        return data;
+      })
+      .catch((error) => {
+        that.emit('error', that, error);
+        that.emit('ready', that, null, error);
+      });
+  }
+
+  _groupSlicesIfMoreThan(data, maxSlices) {
+    if (maxSlices <= 0) {
+      throw RangeError(`Can't create a pie chart with less than 1 slice (asked for ${maxSlices} slices)`);
+    }
+    if (!_.isArray(data.cells) || maxSlices === undefined || data.cells.length <= maxSlices) {
+      return data;
+    }
+
+    const individualCells = data.cells.slice(0, maxSlices - 1);
+    const othersCell = {
+      // Set all dimension's keys that end in "Value" to "Other"
```

Comment:
This is my lack of understanding of OpenSpending's API. I assumed the `keyField` and `keyValue` came from the OpenSpending API, but just noticed that they're added by `babbage.ui` at https://github.com/openspending/babbage.ui/blob/0a7f9eb8006b804112febc6be5c95651a0137957/src/api/index.js#L306-L312. I'll change it.